p2p/transport/webrtc/state.go

```diff
@@ -0,0 +1,70 @@
+package libp2pwebrtc
+
+import (
+	pb "github.com/libp2p/go-libp2p/p2p/transport/webrtc/pb"
+)
+
+type channelState uint8
+
+const (
```

Comment:
not critical but if you turn this into bit flags you can void some of your branches in your methods

as in:

```
const (
   statOpen channelState = 0
   stateReadClosed channelState = 1 << iota
   stateWriteClosed
   stateClosed = stateReadClosed | stateWriteClosed
)
```

Later you can then do stuff like:

```
case pb.Message_FIN:
                return state | stateReadClosed
```

But it does assume that you use it correctly, while your approach is safer in that perspective.